src/foam/nanos/auth/ChangePasswordView.js

```diff
@@ -1,235 +1,231 @@
+      section: 'resetPasswordSection',
+      view: {
+        class: 'net.nanopay.ui.NewPasswordView',
+        passwordIcon: true
+      },
+      minLength: 6
     },
     {
-      class: 'String',
-      name: 'confirmPassword',
-      view: { class: 'foam.u2.view.PasswordView' }
+      class: 'Password',
+      name: 'confirmationPassword',
+      section: 'resetPasswordSection',
+      view: {
+        class: 'foam.u2.view.PasswordView',
+        passwordIcon: true
+      },
+      validationPredicates: [
+        {
+          args: ['newPassword', 'confirmationPassword'],
+          predicateFactory: function(e) {
+            return e.EQ(
+              foam.nanos.auth.ChangePasswordView.NEW_PASSWORD,
+              foam.nanos.auth.ChangePasswordView.CONFIRMATION_PASSWORD);
+          },
+          errorString: 'Passwords do not match.'
+        }
+      ]
     }
   ],
 
+  messages: [
+    { name: 'SUCCESS_MSG', message: 'Your password was successfully updated.' }
+  ],
+
   methods: [
+    {
+      name: 'makeHorizontal',
+      code: function(value) {
+        // evaluation of this gridColumns on a property does not handle an evaluation, only an int, thus function here managing this.
+        this.cls_.getAxiomByName('originalPassword').gridColumns = value ? ( this.token ? 0 : 4 ) : 12;
+        this.cls_.getAxiomByName('newPassword').gridColumns = value ? ( this.token ? 6 : 4 ) : 12;
+        this.cls_.getAxiomByName('confirmationPassword').gridColumns = value ? ( this.token ? 6 : 4 ) : 12;
+      }
+    },
+    {
+      name: 'reset_',
+      code: function() {
+        this.clearProperty('originalPassword');
+        this.clearProperty('newPassword');
+        this.clearProperty('confirmationPassword');
+      }
+    },
     function initE() {
+      this.makeHorizontal(this.horizontal);
       this.SUPER();
       this
-      .addClass(this.myClass())
-      .start().addClass('Container')
-        .start('div')
-          .start('h1').add('Change Password').addClass('changePass-Text').end()
-        .end()
-        .start('div')
-          .start('h2').add('Original Password').addClass('originalPass-Text').end()
-          .start('h2').add('New Password').addClass('newPass-Text').end()
-          .start('h2').add('Confirm Password').addClass('confirmPass-Text').end()
-        .end()
-        .start('div')
-          .start(this.ORIGINAL_PASSWORD).addClass('originalPass-Input').end()
-          .start(this.NEW_PASSWORD).addClass('newPass-Input').end()
-          .start(this.CONFIRM_PASSWORD).addClass('confirmPass-Input').end()
-        .end()
-        .start(this.UPDATE_PASSWORD).addClass('update-BTN').end()
-      .end()
-    }
+        .addClass(this.myClass())
+        .startContext({ data: this })
+          .start().addClass('top-bar').show(this.topBarShow)
+            .start('img')
+              .attr('src', this.theme.logo)
+              .callIfElse( this.horizontal, function() {
```

Comment:
Here's another way you could do this. I'll leave it up to you to decide if it's better.
```suggestion
.enableClass('logoHorizontal', this.horizontal$);
.enableClass('logoCenterVerticle', this.horizontal$, true);
```
The third argument in enableClass negates the evaluated boolean value from the second argument.

Reply:
This would also provide the view the ability toggle between the two states without reinitializing the view.